**Import required packages**

In [60]:
import os
import pandas as pd
import datetime
import seaborn as sns
import matplotlib.pyplot as plt
import numpy as np
%matplotlib inline

**Import DataSet**

In [50]:
path="C:\Local Disk Custom\Edu\Data Science\Springboard\Capstone 2\Ideas and Related datasets\Airline passenger satisfaction"
os.chdir(path) 

filename = 'Airline-Dataset.csv'
df = pd.read_csv(filename)
df.drop(['Unnamed: 0'], axis=1, inplace=True)
df.head()

airline_name        author author_country  \
0  adria-airways         D Ito        Germany   
1  adria-airways  Ron Kuhlmann  United States   
2  adria-airways       E Albin    Switzerland   
3  adria-airways  Tercon Bojan      Singapore   
4  adria-airways       L James         Poland   

                                             content     cabin_flown  \
0  Outbound flight FRA/PRN A319. 2 hours 10 min f...         Economy   
1  Two short hops ZRH-LJU and LJU-VIE. Very fast ...  Business Class   
2  Flew Zurich-Ljubljana on JP365 newish CRJ900. ...         Economy   
3  Adria serves this 100 min flight from Ljubljan...  Business Class   
4  WAW-SKJ Economy. No free snacks or drinks on t...         Economy   

   overall_rating  seat_comfort_rating  cabin_staff_rating  \
0             7.0                  4.0                 4.0   
1            10.0                  4.0                 5.0   
2             9.0                  5.0                 5.0   
3             8.0                  4.0                 4.0   
4             4.0                  4.0                 2.0   

   food_beverages_rating  inflight_entertainment_rating  value_money_rating  \
0                    4.0                            0.0                 4.0   
1                    4.0                            1.0                 5.0   
2                    4.0                            0.0                 5.0   
3                    3.0                            1.0                 4.0   
4                    1.0                            2.0                 2.0   

   Month  Year  recommended  
0      4  2015            1  
1      1  2015            1  
2      9  2014            1  
3      9  2014            1  
4      6  2014            0

**Data Definition**

In [51]:
df.columns

Index(['airline_name', 'author', 'author_country', 'content', 'cabin_flown',
       'overall_rating', 'seat_comfort_rating', 'cabin_staff_rating',
       'food_beverages_rating', 'inflight_entertainment_rating',
       'value_money_rating', 'Month', 'Year', 'recommended'],
      dtype='object')

In [52]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 27284 entries, 0 to 27283
Data columns (total 14 columns):
airline_name                     27284 non-null object
author                           27284 non-null object
author_country                   27284 non-null object
content                          27284 non-null object
cabin_flown                      27284 non-null object
overall_rating                   27284 non-null float64
seat_comfort_rating              27284 non-null float64
cabin_staff_rating               27284 non-null float64
food_beverages_rating            27284 non-null float64
inflight_entertainment_rating    27284 non-null float64
value_money_rating               27284 non-null float64
Month                            27284 non-null int64
Year                             27284 non-null int64
recommended                      27284 non-null int64
dtypes: float64(6), int64(3), object(5)
memory usage: 2.9+ MB


In [53]:
#'cabin_flow' and 'recommended' columns should be converted into categorical data
df = df.astype({"cabin_flown":'category',"recommended":'category'})

In [54]:
#'content' columns contains the review statement which shall not be useful in statistics. So drop it!
df.drop(['content'], axis=1, inplace=True)

In [55]:
#Converting 'Month' and 'Year' columns into a single 'datetime' object
df = df.astype({"Month":'str',"Year":'str'})
df['time'] = df['Month'] + df['Year']
df['time'] = pd.to_datetime(df['time'], format='%m%Y')
df.drop(['Month','Year'], axis=1, inplace=True)
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 27284 entries, 0 to 27283
Data columns (total 12 columns):
airline_name                     27284 non-null object
author                           27284 non-null object
author_country                   27284 non-null object
cabin_flown                      27284 non-null category
overall_rating                   27284 non-null float64
seat_comfort_rating              27284 non-null float64
cabin_staff_rating               27284 non-null float64
food_beverages_rating            27284 non-null float64
inflight_entertainment_rating    27284 non-null float64
value_money_rating               27284 non-null float64
recommended                      27284 non-null category
time                             27284 non-null datetime64[ns]
dtypes: category(2), datetime64[ns](1), float64(6), object(3)
memory usage: 2.1+ MB


**No missing values**
**as each column has 27284 values**

**Drop duplicate rowa (if any)**

In [56]:
print("Old shape:", df.shape)
df = df.drop_duplicates()
print("New shape:", df.shape)

Old shape: (27284, 12)
New shape: (27198, 12)


In [57]:
df.describe()

overall_rating  seat_comfort_rating  cabin_staff_rating  \
count    27198.000000         27198.000000        27198.000000   
mean         6.062027             3.257482            3.520994   
std          3.215240             1.351379            1.459677   
min          1.000000             0.000000            0.000000   
25%          3.000000             2.000000            2.000000   
50%          7.000000             4.000000            4.000000   
75%          9.000000             4.000000            5.000000   
max         10.000000             5.000000            5.000000   

       food_beverages_rating  inflight_entertainment_rating  \
count           27198.000000                    27198.00000   
mean                3.014192                        2.55004   
std                 1.514579                        1.69504   
min                 0.000000                        0.00000   
25%                 2.000000                        1.00000   
50%                 3.000000                        3.00000   
75%                 4.000000                        4.00000   
max                 5.000000                        5.00000   

       value_money_rating  
count        27198.000000  
mean             3.409589  
std              1.440581  
min              0.000000  
25%              2.000000  
50%              4.000000  
75%              5.000000  
max              5.000000

**Data Cleaned**

***Try grouping ratings by airline and class, and derive the Top 10 Highest Rated Airlines!***

In [58]:
avgratings = df.groupby(['airline_name', 'cabin_flown']).mean().dropna()
avgratings.shape

(582, 6)

In [63]:
avgratings.sort_values(by=['overall_rating'], ascending=False).head(10)

overall_rating  seat_comfort_rating  \
airline_name      cabin_flown                                            
yangon-airways    Economy                    10.0                  5.0   
air-vanuata       Business Class             10.0                  3.0   
edelweiss-air     Premium Economy            10.0                  4.0   
egyptair          First Class                10.0                  5.0   
emirates          Premium Economy            10.0                  5.0   
estonian-air      Premium Economy            10.0                  5.0   
shanghai-airlines First Class                10.0                  5.0   
air-zimbabwe      Economy                    10.0                  4.0   
scoot             Premium Economy            10.0                  4.0   
iberia            First Class                10.0                  5.0   

                                   cabin_staff_rating  food_beverages_rating  \
airline_name      cabin_flown                                                  
yangon-airways    Economy                         5.0                    4.0   
air-vanuata       Business Class                  5.0                    4.0   
edelweiss-air     Premium Economy                 5.0                    5.0   
egyptair          First Class                     4.0                    4.0   
emirates          Premium Economy                 5.0                    5.0   
estonian-air      Premium Economy                 5.0                    5.0   
shanghai-airlines First Class                     5.0                    4.0   
air-zimbabwe      Economy                         5.0                    4.0   
scoot             Premium Economy                 5.0                    3.0   
iberia            First Class                     5.0                    4.0   

                                   inflight_entertainment_rating  \
airline_name      cabin_flown                                      
yangon-airways    Economy                                    0.0   
air-vanuata       Business Class                             2.0   
edelweiss-air     Premium Economy                            4.0   
egyptair          First Class                                4.0   
emirates          Premium Economy                            5.0   
estonian-air      Premium Economy                            4.0   
shanghai-airlines First Class                                0.0   
air-zimbabwe      Economy                                    0.0   
scoot             Premium Economy                            4.0   
iberia            First Class                                5.0   

                                   value_money_rating  
airline_name      cabin_flown                          
yangon-airways    Economy                         5.0  
air-vanuata       Business Class                  5.0  
edelweiss-air     Premium Economy                 5.0  
egyptair          First Class                     5.0  
emirates          Premium Economy                 4.0  
estonian-air      Premium Economy                 4.0  
shanghai-airlines First Class                     5.0  
air-zimbabwe      Economy                         3.0  
scoot             Premium Economy                 5.0  
iberia            First Class                     5.0